src/core/connection.h

```diff
@@ -8,27 +8,36 @@ namespace mavsdk {
 
 class Connection {
 public:
-    typedef std::function<void(mavlink_message_t& message)> receiver_callback_t;
+    typedef std::function<void(mavlink_message_t& message, Connection* connection)>
+        receiver_callback_t;
 
-    Connection(receiver_callback_t receiver_callback);
+    Connection(
+        receiver_callback_t receiver_callback,
+        ForwardingOption forwarding_option = ForwardingOption::ForwardingOff);
     virtual ~Connection();
 
     virtual ConnectionResult start() = 0;
     virtual ConnectionResult stop() = 0;
 
     virtual bool send_message(const mavlink_message_t& message) = 0;
 
+    bool do_forward_messages() const;
+    uint8_t forwarding_connections_count() const;
```

Comment:
```suggestion
    unsigned forwarding_connections_count() const;
```

Sorry for the confusion.